libsolc/libsolc.cpp

```diff
@@ -38,6 +40,24 @@ using namespace solidity;
 namespace
 {
 
+// The strings in this list must not be reiszed after they have been added here (via solidity_alloc()), because
+// this may potentially change the pointer that was passed to the caller from solidity_alloc().
+static list<string> solidity_allocations;
+
+/// Find the equivalent to @p _data in the list of allocations of solidity_alloc(),
+/// removes it from the list and returns its value.
+///
+/// If any invalid argument is being passed, it is considered a programming error
+/// on the caller-side and hence, will call abort() then.
+string take_over_allocation(char const* _data)
+{
+	auto iter = find_if(begin(solidity_allocations), end(solidity_allocations), [=](string& _chunk) { return _chunk.data() == _data; });
+	if (iter != end(solidity_allocations))
+		return move(*iter);
```

Comment:
Does that remove it from the linked list?

Reply:
I'm not sure, now that I see this question (guess not), good catch. Made sure now.